src/ZoneServer/Skills/Handlers/Archer/TwinArrows.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Threading.Tasks;
+using Melia.Shared.L10N;
+using Melia.Shared.Tos.Const;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.World.Actors;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Skills.Handlers.Archer
+{
+	/// <summary>
+	/// Handles the Archer skill TwinArrows.
+	/// </summary>
+	[SkillHandler(SkillId.Archer_TwinArrows)]
+	public class TwinArrows : ITargetSkillHandler
+	{
+		/// <summary>
+		/// Handles the skill, do two consecutive hits on the enemy.
+		/// </summary>
+		public void Handle(Skill skill, ICombatEntity caster, ICombatEntity designatedTarget)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.TurnTowards(designatedTarget);
+
+			if (designatedTarget == null)
+			{
+				return;
+			}
+
+			if (!caster.Position.InRange2D(designatedTarget.Position, skill.Data.MaxRange))
+			{
+				caster.ServerMessage(Localization.Get("Too far away."));
+				return;
+			}
+
+			this.Attack(skill, caster, designatedTarget);
+		}
+
+		/// <summary>
+		/// Executes the actual attack.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="designatedTarget"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ICombatEntity designatedTarget)
+		{
+			var damageDelay = TimeSpan.Zero;
+			var skillHitDelay = skill.Data.DefaultHitDelay;
+
+			// TODO: Add more 50% damage to enemies using cloth armor type
+
+			var skillHitResult = SCR_SkillHit(caster, designatedTarget, skill);
+			designatedTarget.TakeDamage(skillHitResult.Damage, caster);
+
+			var skillHit = new SkillHitInfo(caster, designatedTarget, skill, skillHitResult, damageDelay, skillHitDelay);
+
+			Send.ZC_SKILL_READY(caster, skill, caster.Position, caster.Position);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, caster.Position, null);
```

Comment:
According to my logs this should be `ZC_SKILL_FORCE_TARGET`. What about yours?

Reply:
Well, if I don`t send the ZC_SKILL_READY and ZC_SKILL_MELEE_GROUND before the ZC_SKILL_HIT_INFO, the character does`t correct does the animations + sound effect. If you find a better way feel free to tell me.

Reply:
What I mean is that you presumably shouldn't be sending `ZC_SKILL_MELEE_GROUND` but `ZC_SKILL_FORCE_TARGET`.

Reply:
From my tests on the server, it seems like ZC_SKILL_FORCE_TARGET is not being received, altough ZC_SKILL_HIT_INFO is,
but for it to work we should use the other two packets that I mentioned.
I am getting redundant but I guess you get it.

I am gonna try to simple switch to ZC_SKILL_FORCE_TARGET to see if I get a better result.